pom.xml

```diff
@@ -129,6 +129,11 @@
             <version>1.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache-extras.beanshell</groupId>
```

Comment:
Why not use aviator, the same evaluator for matcher?

Reply:
Using this framework makes the code easier to implement and more powerful, because it simulates Java statements. So the expression like `r.sub.name.equals("Alice")` will also be recognized. In fact, you can write any boolean expression that conforms to the Java syntax specification here.